core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/openapi.kt

```diff
@@ -0,0 +1,2173 @@
+                                run {
+                                    val additionalPropertyPaths = (marker as OpenApiMarker)
+                                        .additionalPropertyPaths
+                                        .distinct()
+
+                                    "return listOf(${additionalPropertyPaths.joinToString { "JsonPath(\"\"\"${it.path}\"\"\")" }})"
+                                }
+                            )
+                            .build()
+                    )
+                    .build()
+            )
+            .addFunction(
+                FunSpec.builder("readJson")
+                    .returns(returnType)
+                    .addParameter("url", URL::class)
+                    .addCode(getReadAndConvertMethod("readJson(url, typeClashTactic = ANY_COLUMNS, keyValuePaths = keyValuePaths)"))
+                    .build()
+            )
+            .addFunction(
+                FunSpec.builder("readJson")
+                    .returns(returnType)
+                    .addParameter("path", String::class)
+                    .addCode(getReadAndConvertMethod("readJson(path, typeClashTactic = ANY_COLUMNS, keyValuePaths = keyValuePaths)"))
+                    .build()
+            )
+            .addFunction(
+                FunSpec.builder("readJson")
+                    .returns(returnType)
+                    .addParameter("stream", InputStream::class)
+                    .addCode(getReadAndConvertMethod("readJson(stream, typeClashTactic = ANY_COLUMNS, keyValuePaths = keyValuePaths)"))
+                    .build()
+            )
+            .addFunction(
+                FunSpec.builder("readJsonStr")
+                    .returns(returnType)
+                    .addParameter("text", String::class)
+                    .addCode(getReadAndConvertMethod("readJsonStr(text, typeClashTactic = ANY_COLUMNS, keyValuePaths = keyValuePaths)"))
+                    .build()
+            )
+            .build()
+
+        return typeSpec.toString()
+    }
+}
+
+/** Needs to have any type schemas to convert. */
+internal fun isOpenApiStr(text: String): Boolean {
+    val parsed = OpenAPIParser().readContents(text, null, null)
+    return parsed.openAPI?.components?.schemas != null
+}
+
+internal fun isOpenApi(path: String): Boolean = isOpenApi(asURL(path))
+
+internal fun isOpenApi(url: URL): Boolean {
+    if (url.path.endsWith(".yml") || url.path.endsWith("yaml")) {
+        return true
+    }
+    if (!url.path.endsWith("json")) {
+        return false
+    }
+
+    return isOpenApiStr(url.readText())
+}
+
+internal fun isOpenApi(file: File): Boolean {
+    if (file.extension.lowercase() in listOf("yml", "yaml")) {
+        return true
+    }
+
+    if (file.extension.lowercase() != "json") {
+        return false
+    }
+
+    return isOpenApiStr(file.readText())
+}
+
+/** Parse and read OpenApi specification to [DataSchema] interfaces. */
+public fun readOpenApi(
```

Comment:
why not make DataFrame.Companion receiver?

Reply:
Because you don't create a `DataFrame` ;)